src/main/kotlin/org/wycliffeassociates/otter/jvm/utils/TornadoFXUtils.kt

```diff
@@ -13,3 +17,33 @@ fun <T> ObservableValue<T>.onChangeAndDoNow(op: (T?) -> Unit) {
         op(it)
     }
 }
+
+/**
+ * Calls tornadofx's [observeOnFx] if JavaFX is running. Using this method instead of [observeOnFx] directly
+ * makes it safe to run unit tests when the FX platform is not running.
+ */
+fun <T> Observable<T>.observeOnFxSafe(): Observable<T> {
+    return when (IsFXInitialized.isInitialized()) {
+        true -> observeOnFx()
+        false -> this
+    }
+}
+
+private object IsFXInitialized {
+    private var mIsInitialized: Boolean? = null
```

Comment:
why are you using hungarian notation here?

Reply:
Besides the fact that I should use an underscore, there are two other ways I could have implemented this. The first is like this (It is not good, I am only including it because it turned into an interesting thought exercise):

```
private object IsFXInitialized {
    private var _isInitialized: Boolean? = null
        get() {
            if (field == null) {
                field = testIsInitialized()
            }
            return field
        }
    fun isInitialized(): Boolean = _isInitialized!!

    private fun testIsInitialized(): Boolean {
        try {
            Platform.runLater{}
        } catch (e: IllegalStateException) {
            return false
        }
        return true
    }
}
```

Obviously we probably don't like this because there's still a naming issue, and now we need a `!!` operator. 

The second way is like this, which is what I ended up going with and committing just now.

```
private object IsFXInitialized {
    // Since object declarations are initialized lazily, isInitialized will be computed when
    // IsFXInitialized is accessed for the first time
    val isInitialized: Boolean = testIsInitialized()

    private fun testIsInitialized(): Boolean {
        try {
            Platform.runLater{}
        } catch (e: IllegalStateException) {
            return false
        }
        return true
    }
}
```